### PR TITLE
Add basic commit backfill support

### PR DESCRIPTION
**Pull request by mitsuhiko**
This adds missing commits when you associate releases.  This requires the change
from sentry master that adds support for this API (https://github.com/getsentry/sentry/pull/5212)

This comes up particularly during local development when webhooks are not set up
or if someone just starts using sentry-cli with data from before webhooks
were firing.

Since not all data is available in the diffs we cannot add files associated
and similar things without another HTTP request but this is good enough
for now I think.

### CODE REVIEW

File: src/sentry_plugins/github/endpoints/webhook.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def __call__(self, organization, event):
 
             try:
                 with transaction.atomic():
-                    c = Commit.objects.create(
+                    c, created = Commit.objects.get_or_create(
                         repository_id=repo.id,
                         organization_id=organization.id,
                         key=commit['id'],
```
Comment (macqueen) [inline src/sentry_plugins/github/endpoints/webhook.py]: do you not want `message`, `author`, `date_added` in `defaults`? also, is there a reason you're not using `create_or_update`?

File: src/sentry_plugins/github/plugin.py [hunk 2]
```diff
@@ -35,6 +38,44 @@
...
+        if author.name != commit['commit']['author']['name']:
+            author.update(name=commit['commit']['author']['name'])
+
+    # We do not want to update here because the information that comes
+    # from the webhook is of higher quality
+    Commit.objects.create_or_update(
+        repository_id=repo.id,
+        key=commit['sha'],
+        organization_id=repo.organization_id,
+        values=dict(
```
Comment (macqueen) [inline src/sentry_plugins/github/plugin.py]: based on your comment, do you want this to be defaults? https://github.com/getsentry/sentry/blob/master/src/sentry/db/models/query.py#L58. also, if you don't want to update, is there a reason you're not just creating and handling the integrity error?

File: src/sentry_plugins/github/endpoints/webhook.py (hunk 1 above)
Comment (mitsuhiko) [inline src/sentry_plugins/github/endpoints/webhook.py]: @macqueen create_or_update seems to not return the record. I might have been doing something wrong though.

File: src/sentry_plugins/github/plugin.py (hunk 2 above)
Comment (mitsuhiko) [inline src/sentry_plugins/github/plugin.py]: No reason really I suppose. I just made something that works as a proof of concept I guess.

**Comment by mitsuhiko** [comment]
This is dead.

**Pull request closed without merging**